lib/spack/spack/main.py

```diff
@@ -470,7 +470,10 @@ def setup_main_options(args):
         spack.config.add(fullpath=config_var, scope="command_line")
 
     # when to use color (takes always, auto, or never)
-    color.set_color_when(args.color)
+    if args.color == "auto" and "SPACK_COLOR" in os.environ:
+        color.set_color_when(os.environ["SPACK_COLOR"])
+    else:
+        color.set_color_when(args.color)
```

Comment:
I'm re-reading this and I realize it doesn't distinguish between the case of having `--color=auto` set explicitly or not if `SPACK_COLOR` is defined. Maybe a better implementation would just be:
```python
parser.add_argument(
    '--color', action='store', 
    default=os.environ.get('SPACK_COLOR', 'auto'),
    choices=('always', 'never', 'auto'),
    help="when to colorize output (default: SPACK_COLOR value or auto)"
)
```
and remove the modifications here.

Reply:
yeah, that's actually a neat solution